linera-client/src/unit_tests/chain_listener.rs

```diff
@@ -220,8 +220,8 @@ async fn test_chain_listener_admin_chain() -> anyhow::Result<()> {
         .unwrap();
     for i in 0.. {
         linera_base::time::timer::sleep(Duration::from_secs(i)).await;
-        let result = storage.read_certificate(certificate.hash()).await;
-        if result.ok().as_ref() == Some(&certificate) {
+        let result = storage.read_certificate(certificate.hash()).await?;
+        if result == Some(certificate.clone()) {
```

Comment:
```suggestion
        if result.as_ref() == Some(&certificate) {
```

(Or `is_some_and`, if you prefer.)

Reply:
Ok, thanks.